src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1316,13 +1318,17 @@ public static void dispatchMutationsWithRetryOnDifferentSystem(List<? extends IM
                 {
                     if (accordResult != null)
                     {
-                        TxnResult.Kind kind = accordResult.awaitAndGet().kind();
+                        TxnResult result = accordResult.awaitAndGet();
+                        TxnResult.Kind kind = result.kind();
                         if (kind == retry_new_protocol && failure == null)
                         {
                             Tracing.trace("Accord returned retry new protocol");
                             logger.debug("Retrying mutations on different system because some mutations were misrouted according to Accord");
                             continue;
                         }
+                        if (result.kind() == validation_rejection)
+                            throw ((TxnValidationRejection) result).validationException;
+
```

Comment:
Similar if condition checks are present in several parts of the code. Can this be moved to a helper in `TransactionResult` to avoid code duplication?